Source/portal.cpp

```diff
@@ -18,12 +18,24 @@ void InitPortals()
 
 void SetPortalStats(int i, BOOL o, int x, int y, int lvl, int lvltype)
 {
+	#ifndef HELLFIRE
 	portal[i].setlvl = FALSE;
 	portal[i].x = x;
 	portal[i].y = y;
 	portal[i].open = o;
 	portal[i].level = lvl;
 	portal[i].ltype = lvltype;
+	#endif

```

Comment:
Use an `#else` instead.

Are you sure that reordering the variables will break Diablo 1.09? Probably the new order was actually the correct one for both.

Reply:
I was actually wondering about this. I checked diablo and they are differnt . It looks like they did indeed swap the variables. I will post screens .

Reply:
The order for hellfire also matches the beta, so should work in 1.09